examples/rn-connection-and-error/frontend/app/screens/StoreScreen.tsx

```diff
@@ -78,79 +73,76 @@ export function StoreScreen() {
-                  }}
-                  text="Trigger Store Change"
-                />
-              </>
-            )}
-            <Button
-              extraStyles={[styles.button]}
-              onPress={refreshAccessToken}
-              text="Refresh Access Token / User Data"
-            />
-            <Button
-              extraStyles={[styles.button]}
-              onPress={() => {
-                refreshSession();
-              }}
-              text="Refresh Session"
-            />
+          <View style={styles.triggers}>
+            <View style={styles.status}>
+              <Text style={styles.statusText}>
+                Status: {isConnected ? 'Connected 🟢' : 'Not connected 🔴'}
+              </Text>
+              <Button
+                isSecondary
+                onPress={isConnected ? disconnect : reconnect}
+                text={isConnected ? 'Disconnect' : 'Connect'}
+              />
+            </View>
+            <View style={styles.triggerButtons}>
+              <Button
+                extraStyles={[styles.button]}
+                onPress={addKiosk}
+                text="Add Kiosk"
+              />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={
+                  store?.kiosks.length
+                    ? addProduct
+                    : () => Alert.alert('Add a kiosk first.')
+                }
+                text="Add Product"
+              />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={triggerSyncError}
+                text="Trigger Sync Error"
+              />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={triggerClientReset}
+                text="Trigger Client Reset"
+              />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={deleteUser}
+                text="Delete User"
+              />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={() => {
+                  switchStore();
+                  Alert.alert(
+                    'The associated store has been changed. Click "Refresh User Data" to update the UI.',
+                  );
+                }}
+                text="Trigger Store Change"
+              />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={refreshAccessToken}
+                text="Refresh Access Token / User Data"
+              />
+              <Button
+                extraStyles={[styles.button]}
+                onPress={refreshSession}
+                text="Refresh Session"
+              />
+            </View>
           </View>
+        </>
+      ) : (
+        // Store isn't yet loaded, perhaps while switching stores
+        <View style={styles.createStore}>
+          <Loading />
         </View>
-      </>
+      )}
```

Comment:
Made some updates to the layout here since the UI was flickering a bit when switching the store. Or was the restructuring due to issues in refreshing the session by the way?

Reply:
The goal was to keep visible the buttons that may need to be called to ensure the client reset is properly handled.  For instance, if one were to refresh the session and not update the current user data, then we will be stuck in a loading screen and not able to move forward.

Reply:
Thanks for clarifying, updates comin'